packages/contracts/test/exchange/core.ts

```diff
@@ -628,14 +629,14 @@ describe('Exchange', () => {
             expect(newBalances).to.be.deep.equal(balances);
         });
 
-        it.skip('should throw if maker allowances are too low to fill order and \
+        it('should throw if maker allowances are too low to fill order and \
                 shouldThrowOnInsufficientBalanceOrAllowance = true', async () => {
             await rep.approve.sendTransactionAsync(tokenTransferProxy.address, new BigNumber(0), { from: maker });
-            expect(
+            await expectRevertOrAlwaysFailingTransaction(
```

Comment:
Previously, there was a really subtle bug here. Without `await`, if the promise returned by `expect` rejects, it ***does not*** result in a test failure. Instead, it results in a warning being logged about an unhandled promise rejection (which whizzes by and is really easy to miss).

This is a really unfortunate footgun in JavaScript, but it is somewhat possible to guard against it by [making Node throw when it detects unhandled promise rejections](https://medium.com/@dtinth/making-unhandled-promise-rejections-crash-the-node-js-process-ffc27cfcc9dd). I hope to address this in a separate PR.

Reply:
https://github.com/0xProject/0x-monorepo/pull/597 will help us catch cases like this.